modules/kargoAnalyticsAdapter.js

```diff
@@ -1,14 +1,96 @@
+import { logError } from '../src/utils.js';
+import { ajax } from '../src/ajax.js';
 import adapter from '../src/AnalyticsAdapter.js';
 import adapterManager from '../src/adapterManager.js';
+import CONSTANTS from '../src/constants.json';
 
-var kargoAdapter = adapter({
-  analyticsType: 'endpoint',
-  url: 'https://krk.kargo.com/api/v1/event/prebid'
-});
+const EVENT_URL = 'https://krk.kargo.com/api/v1/event';
+const KARGO_BIDDER_CODE = 'kargo';
+
+const analyticsType = 'endpoint';
+
+let _initOptions = {};
+
+let _logBidResponseData = {
+  auctionId: '',
+  auctionTimeout: 0,
+  responseTime: 0,
+};
+
+let _bidResponseDataLogged = [];
+
+var kargoAnalyticsAdapter = Object.assign(
+  adapter({ analyticsType }), {
+    track({ eventType, args }) {
+      switch (eventType) {
+        case CONSTANTS.EVENTS.AUCTION_INIT: {
+          _logBidResponseData.auctionTimeout = args.timeout;
+          break;
+        }
+        case CONSTANTS.EVENTS.BID_RESPONSE: {
+          handleBidResponseData(args);
+          break;
+        }
+      }
+    }
+  }
+);
+
+// handleBidResponseData: Get auction data for Kargo bids and send to server
+function handleBidResponseData (bidResponse) {
+  // Verify this is Kargo and we haven't seen this auction data yet
+  if (bidResponse.bidder !== KARGO_BIDDER_CODE || _bidResponseDataLogged.includes(bidResponse.auctionId) !== false) {
```

Comment:
Two things:

1. What if somebody is using an alias name for Kargo, do you still want to track it or not? If so, you can look at the aliasRegistry in order to determine if a bidder is an alias of kargo or not.


2. This array of auctionId's definitely works. I only suggested a MAP because you can do something like:

```js
const _bidResponseDataLogged = {
  auctionId1: true,
  auctionId2: false
}
```
If NOT (data has sent AKA map at auctionId is true) 
```js
if (!_bidResponseDataLogged[bidResponse.auctionId]) {
   // log data
}
```

Which is a O(1) lookup.

But honestly, includes is prob fine as the array will not get too big for a given page load for most integrations!